jumanji/environments/routing/mmst/env.py

```diff
@@ -0,0 +1,721 @@
+        Each agent selects the next node to which it wants to connect.
+
+    - state: State
+        - node_type: jax array (int) of shape (num_nodes,).
+            the component type of each node (-1 represents utility nodes).
+        - adj_matrix: jax array (bool) of shape (num_nodes, num_nodes):
+            adjacency matrix of the graph.
+        - connected_nodes: jax array (int) of shape (num_agents, step_limit).
+            we only count each node visit once.
+        - connected_nodes_index: jax array (int) of shape (num_agents, num_nodes).
+        - position_index: jax array (int) of shape (num_agents,).
+        - node_edges: jax array (int) of shape (num_agents, num_nodes, num_nodes).
+        - positions: jax array (int) of shape (num_agents,).
+            the index of the last visited node.
+        - action_mask: jax array (bool) of shape (num_agent, num_nodes).
+            binary mask (False/True <--> invalid/valid action).
+        - finished_agents: jax array (bool) of shape (num_agent,).
+        - nodes_to_connect: jax array (int) of shape (num_agents, num_nodes_per_agent).
+        - step_count: step counter.
+        - step_limit: the number of steps allowed before an episode terminates.
+        - key: PRNG key for random sample.
+
+    - constants definitions:
+        - Nodes
+            - INVALID_NODE = -1: used to check if an agent selects an invalid node.
+                A node may be invalid if its has no edge with the current node or if it is a
+                utility node already selected by another agent.
+            - UTILITY_NODE = -1: utility node (belongs to no agent).
+            - EMPTY_NODE = -1: used for padding.
+                state.connected_nodes stores the path (all the nodes) visited by an agent. Hence
+                it has size equal to the step limit. We use this constant to initialise this array
+                since 0 represents the first node.
+            - DUMMY_NODE = -10: used for tie-breaking if multiple agents select the same node.
+
+        - Edges
+            - EMPTY_EDGE = -1: used for masking edges array.
+               state.node_edges is the graph's adjacency matrix, but we don't represent it
+               using 0s and 1s, we use the node values instead, i.e `A_ij = j` or `A_ij = -1`.
+               Also edges are masked when utility nodes
+               are selected by an agent to make it unaccessible by other agents.
+
+        - Actions encoding
+            - INVALID_CHOICE = -1
+            - INVALID_TIE_BREAK = -2
+            - INVALID_ALREADY_TRAVERSED = -3
+    """
+
+    def __init__(
+        self,
+        generator_fn: Optional[Generator] = None,
+        reward_fn: Optional[RewardFn] = None,
+        step_limit: int = 100,
+        viewer: Optional[Viewer[State]] = None,
+    ):
+        """Create the `MMST` environment.
+
+        Args:
+            generator_fn: `Generator` whose `__call__` instantiates an environment instance.
```

Comment:
It would be easily understandable if we have the parameters the default `generator_fn` is inistantiated with in the docstring